src/context/modal-context.tsx

```diff
@@ -32,6 +34,7 @@ const ModalProvider: FC<ModalProviderProps> = ({ children }) => {
   const [modal, setModal] = useState<ReactElement | null>(null)
   const [paperProps, setPaperProps] = useState<PaperProps>({})
   const [timer, setTimer] = useState<NodeJS.Timeout | null>(null)
+  const [customCloseModal, setCustomCloseModal] = useState<() => void>()
```

Comment:
Please consider using `useRef` to store `customCloseModal` instead of `useState` to avoid unnecessary rerenders in context.

Reply:
customCloseModal is stored inside of modal-context and is a part of close modal behavior, when user click on close icon
When I've changed useState into useRef (for testing) context still rerenders each time, when we click on close icon.

Now, handleCloseModal function looks like that

![image](https://github.com/ita-social-projects/SpaceToStudy-Client/assets/127191734/35b75e56-7958-4acc-9ff0-813af6504fe0)

Reply:
I meant that you have an extra rerender on init phase in following code:

```jsx
paperProps && setPaperProps(paperProps)
delayToClose && closeModalAfterDelay(delayToClose)
customCloseModal && setCustomCloseModal(() => customCloseModal)
```

setState always causes rerenders, so `setCustomCloseModal` will trigger a rerender. If component still rerenders, I think using `useRef` is a better option